### PR TITLE
New tickers on deletion compactions grouped by reasons

### PR DESCRIPTION
**Pull request by hx235**
Context/Summary: as titled

Test plan: piggyback on existing tests; fixed a failed test due to adding new stats

### CODE REVIEW

File: java/rocksjni/portal.h [hunk 1]
```diff
@@ -5528,6 +5534,14 @@ class TickerTypeJni {
         return ROCKSDB_NAMESPACE::Tickers::TABLE_OPEN_PREFETCH_TAIL_HIT;
       case -0x3C:
         return ROCKSDB_NAMESPACE::Tickers::BLOCK_CHECKSUM_MISMATCH_COUNT;
+      case -0x3D:
+        return ROCKSDB_NAMESPACE::Tickers::READAHEAD_TRIMMED;
```
Comment (hx235) [inline java/rocksjni/portal.h:5538]: This existing stat missed java conversion here previously

**Review by cbi42 — APPROVED** [review]
LGTM!

**Comment by hx235** [comment]
Updated the tickers name as suggested by @ajkr

File: db/compaction/compaction_picker_fifo.cc [hunk 2]
```diff
@@ -443,9 +446,32 @@ Compaction* FIFOCompactionPicker::PickCompaction(
...
+      break;
+    case CompactionReason::kFIFOTtl:
+      RecordTick(statistics, FIFO_TTL_COMPACTIONS);
+      break;
+    case CompactionReason::kChangeTemperature:
+      RecordTick(statistics, FIFO_CHANGE_TEMPERATURE_COMPACTIONS);
+      break;
+    default:
+      break;
+  }
```
Comment (ajkr) [inline db/compaction/compaction_picker_fifo.cc:472]: Based on the names I would have guessed only completed compactions count. Is it difficult to make it work that way?

File: include/rocksdb/statistics.h [hunk 3]
```diff
@@ -518,6 +518,11 @@ enum Tickers : uint32_t {
   // ReadOptions.auto_readahead_size is set.
   READAHEAD_TRIMMED,
 
+  // Number of FIFO compactions that drop files based on different reasons
+  FIFO_MAX_SIZE_COMPACTIONS,
+  FIFO_TTL_COMPACTIONS,
+  FIFO_CHANGE_TEMPERATURE_COMPACTIONS,
```
Comment (ajkr) [inline include/rocksdb/statistics.h:524]: Is "drop files" an accurate way to describe this type of compaction? I thought it's a time-triggered intra-L0 that rewrites data into the next tier

Comment (hx235) [reply include/rocksdb/statistics.h]: You are correct - we should not include FIFO_CHANGE_TEMPERATURE_COMPACTIONS as deletion compaction. Removed ticker for it.

File: db/compaction/compaction_picker_fifo.cc (hunk 2 above)
Comment (hx235) [reply db/compaction/compaction_picker_fifo.cc]: For deletion compaction (the scope of this PR), it's not difficult. Fixed. Thank you!

**Review by ajkr — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**